### ISSUE TITLE
Build the official Mac release with Qt

### ISSUE THREAD

**Issue by cspiegel** [garglk/garglk#904]
Both Unix and Windows now use Qt for their interface. Mac still uses Cocoa, but it would be ideal to switch to Qt on Mac as well to avoid duplication.

At the moment, there are two major sticking points with Qt on Mac:

1. The UI is not Mac-like; it acts more like a normal Unix or Windows program. You can't launch new instances, and closing the window closes the program. These basic operations will need to be added to the Mac interface before it can be a proper replacement.
2. I've never been able to bundle the Qt version, due to library issues (I can add details to this next time I try it).

Once these are addressed, the Cocoa code can be removed entirely so that there is a unified UI.

**Comment by angstsmurf** [comment]
> I was playing Zork 40 years ago and I would really love to have all Infocom adventures (including Zork Zero) playing like they used to ....

Perhaps off-topic, but in case you weren't aware, there is an Emscripten version of Frotz that lets you play _Zork Zero_ and the other graphical Infocom games in your browser.
https://curiousdannii.github.io/infocom-frotz/
https://curiousdannii.github.io/infocom-frotz/zorkzero.html

**Comment by cspiegel** [comment]
> Hi all, IMHO it is less important that the UI is not fully mac-like in this case and it is more important to have a single code base where bugs are fixed simultaneously. I was playing Zork 40 years ago and I would really love to have all Infocom adventures (including Zork Zero) playing like they used to ....

The way the games play is (or should be) identical on either platform; it's the "scaffolding", so to speak, that differs. I agree that a unified codebase is ideal, but not at the expense of the UI that people are used to, or expect, on one of the platforms.

If there are cases where games don't play properly on Mac but do on Qt platforms, please do let me know; but given how Gargoyle works, that's generally just not going to be an issue.

In short, Gargoyle uses the same code on all platforms to play and render games to a canvas, and the UI code primarily just displays that canvas, and takes user input. So anything game related is all done _before_ the UI is involved.
